NX 21.2 - ng-packagr failure with angular 18 [Cannot find module 'ng-packagr/src/lib/utils/color']

### Current Behavior

seems like there was an unintended breaking change with 21.2. 
[the docs seems to state that angular 17 is not supported anymore](https://nx.dev/blog/nx-21-2-release#breaking-change-angular-17-support-removed), but it seems like 18 support got impaired.

all compilation now fail on this error
```
 NX   Cannot find module 'ng-packagr/src/lib/utils/color'

Require stack:
- /node_modules/@nx/angular/src/executors/utilities/ng-packagr/pre-v19/stylesheet-processor.js
- /node_modules/@nx/angular/src/executors/utilities/ng-packagr/stylesheet-processor.di.js: Error during instantiation of InjectionToken ng.v5.stylesheetProcessor! (InjectionToken ng.v5.packageTransform -> InjectionToken ng.v5.entryPointTransform -> InjectionToken ng.v5.compileNgcTransform -> InjectionToken ng.v5.stylesheetProcessor).
Pass --verbose to see the stacktrace.

```


### Expected Behavior

should import without issues from `ng-packagr/lib/utils/color`

### GitHub Repo

_No response_

### Steps to Reproduce

1. upgrade to nx 21.2
2. build with ng-packagr
3. failure

### Nx Report

```shell
NX   Report complete - copy this into the issue template

Node           : 20.17.0
OS             : darwin-arm64
Native Target  : aarch64-macos
npm            : 10.8.2

nx                 : 21.2.0
@nx/js             : 21.2.0
@nx/jest           : 21.2.0
@nx/eslint         : 21.2.0
@nx/workspace      : 21.2.0
@nx/angular        : 21.2.0
@nx/cypress        : 21.2.0
@nx/devkit         : 21.2.0
@nx/eslint-plugin  : 21.2.0
@nx/plugin         : 21.2.0
@nx/react          : 21.2.0
@nx/rspack         : 21.2.0
@nx/storybook      : 21.2.0
@nx/vite           : 21.2.0
typescript         : 5.5.4

---------------------------------------
Nx key licensed packages

@nx/gcs-cache  : 1.3.1

---------------------------------------
Community plugins:
@analogjs/platform            : 1.17.1
@analogjs/vite-plugin-angular : 1.17.1
@compodoc/compodoc            : 1.1.26
@nx/gcs-cache                 : 1.3.1
@spartan-ng/cli               : 0.0.1-alpha.404
---------------------------------------
Local workspace plugins:
@pan/nx
@pan/cas
---------------------------------------
Cache Usage: 335.46 MB / 46.04 GB
```

### Failure Logs

```shell
NX   Cannot find module 'ng-packagr/src/lib/utils/color'

Require stack:
- /node_modules/@nx/angular/src/executors/utilities/ng-packagr/pre-v19/stylesheet-processor.js
- /node_modules/@nx/angular/src/executors/utilities/ng-packagr/stylesheet-processor.di.js: Error during instantiation of InjectionToken ng.v5.stylesheetProcessor! (InjectionToken ng.v5.packageTransform -> InjectionToken ng.v5.entryPointTransform -> InjectionToken ng.v5.compileNgcTransform -> InjectionToken ng.v5.stylesheetProcessor).
Pass --verbose to see the stacktrace.
```

### Package Manager Version

_No response_

### Operating System

- [x] macOS
- [ ] Linux
- [ ] Windows
- [ ] Other (Please specify)

### Additional Information

Open PR - https://github.com/nrwl/nx/pull/31598

--- a/packages/angular/src/executors/utilities/ng-packagr/pre-v19/stylesheet-processor.ts
+++ b/packages/angular/src/executors/utilities/ng-packagr/pre-v19/stylesheet-processor.ts
@@ -9,8 +9,9 @@
 import { workspaceRoot } from '@nx/devkit';
 import browserslist from 'browserslist';
 import { existsSync } from 'fs';
-import { colors } from 'ng-packagr/src/lib/utils/color';
 import { dirname, join } from 'path';
+import { getNgPackagrVersionInfo } from '../ng-packagr-version';
+import { importNgPackagrPath } from '../package-imports';
 
 const maxWorkersVariable = process.env['NG_BUILD_MAX_WORKERS'];
 const maxThreads =
@@ -100,6 +101,10 @@ export class StylesheetProcessor {
     ]);
     const tailwindConfigPath = findTailwindConfiguration(searchDirs);
 
+    const { major: ngPackagrMajorVersion } = getNgPackagrVersionInfo();
+    const { colors } = importNgPackagrPath<
+      typeof import('ng-packagr/src/lib/utils/color')
+    >('ng-packagr/src/lib/utils/color', ngPackagrMajorVersion);
     const Piscina = getPiscina();
 
     this.renderWorker = new Piscina({

